text/0000-vhdl-emitter.md

```diff
@@ -0,0 +1,138 @@
+
+Nonetheless, it is likely easier to get all the VHDL emitter `Transform`s integrate with the way things work now and then refactor the emitter to be a separate stage.
+
+# Prior art
+[prior-art]: #prior-art
+
+For prior art, the existing Verilog emitter is likely the best example of how to do this.
+
+SpinalHDL seems to do just fine with maintaining two emitters, both Verilog and [spinal.core.internals.PhaseVhdl](https://github.com/SpinalHDL/SpinalHDL/blob/dev/core/src/main/scala/spinal/core/internals/PhaseVhdl.scala). SpinalHDL also seems to do a better job of giving users what they want and has some seeming traction in Europe where VHDL is more prevalent. From a "give the users what they want" perspective, there's good reason to add a VHDL back-end.
+
+Additionally, there is a fair amount of similarity related to the old Chisel2 implementation of both a C++ emitter and a Verilog emitter. The initial motivation for this was that Verilator, at the time, was not as fast as it is now and Berkeley could do better with their own implementation. As Verilator improved, it made no sense to maintain an inferior backend that introduced questions of equivalence. A VHDL back-end is very similar in that it's expected to be less performant in simulation by open source tools (not it's fault, but a problem with missing tools) and has equivalence difficulties with Verilog.
+
+Anecdotally, maintenance of the C++ and Verilog backends in Chisel2 was a known headache and moving to a single emitter in Chisel3 was a good approach.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- Should this be implemented as a separate `Stage` or should we follow the model of the existing Verilog emitter family?
+- Is there enough motivation to add a VHDL back-end and do users actually want this?
+- What resources do we have in place to maintain this and how can we go about verifying that both the Verilog and VHDL back-ends are equivalent?
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+This RFC is the first instance of adding a *new* back-end to the FIRRTL compiler. This motivates two future directions:
+
+- This should be a model for how new back-ends should be added.
+- This should be a starting point for thinking about what back-end `Stage`s should look like. This would either be part of this RFC as it evolves or it would be part of a future RFC about building back-end stages.
+
+It would be beneficial if we had a dedicated owner of the VHDL back-end and if we had dedicated VHDL users. Without either of these, a VHDL back-end is likely to diverge and become difficult to maintain. Granted, we can't have VHDL users now if we don't emit VHDL. Any users would be mixed-language ones.
```

Comment:
It may be possible to check Verilog/VHDL for semantic equivalence using [hdlConvertor](https://github.com/Nic30/hdlConvertor), current state of hdlConvertor is enough for the needs for FIRRTL (parses and emits VHDL/Verilog from/to same AST), but it currently does not support 100% of VHDL2008/SystemVerilog2017 (+-0.5y).